src/Sylius/Bundle/ApiBundle/Filters/SearchFilter.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Filters;
+
+use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\SearchFilter as BaseSearchFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\QueryBuilder;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+class SearchFilter extends BaseSearchFilter
+{
+    public function __construct(
+        ManagerRegistry $managerRegistry,
+        ?RequestStack $requestStack,
+        IriConverterInterface $iriConverter,
+        PropertyAccessorInterface $propertyAccessor = null,
+        LoggerInterface $logger = null,
+        array $properties = null,
+        IdentifiersExtractorInterface $identifiersExtractor = null,
+        NameConverterInterface $nameConverter = null
+    ) {
+        parent::__construct(
+            $managerRegistry,
+            $requestStack,
+            $iriConverter,
+            $propertyAccessor,
+            $logger,
+            $properties,
+            $identifiersExtractor,
+            $nameConverter
+        );
+    }
+
+    protected function filterProperty(
+        string $property,
+        $value,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ) {
+        parent::filterProperty($property, $value, $queryBuilder, $queryNameGenerator, $resourceClass, $operationName);
+    }
+
+    public function getDescription(string $resourceClass): array
+    {
+        return parent::getDescription($resourceClass);
+    }
+
+    protected function addWhereByStrategy(
```

Comment:
It seems like extending a base `SearchFilter` gives us more questions than answers 😄 Can we try the [custom filter](https://api-platform.com/docs/core/filters/#creating-custom-doctrine-orm-filters) approach?